apps/core/components/nav-link/index.tsx

```diff
@@ -0,0 +1,34 @@
+'use client';
+
+import { ComponentProps, ElementRef, forwardRef } from 'react';
+
+import { cn } from '~/lib/utils';
+
+import { Link } from '../../navigation';
+
+type NavLinkProp = ComponentProps<typeof Link>;
+type NavLinkType = Omit<React.AnchorHTMLAttributes<HTMLAnchorElement>, keyof NavLinkProp> &
+  NavLinkProp & {
+    children?: React.ReactNode;
+  } & React.RefAttributes<HTMLAnchorElement>;
+
+// This component wraps next/link and automatically prefixes the href with the current locale as necessary.
+// https://next-intl-docs.vercel.app/docs/routing/navigation#link
+export const NavLink = forwardRef<ElementRef<'a'>, NavLinkType>(
+  ({ href, prefetch = false, children, className, ...rest }, ref) => {
+    return (
+      <Link
+        className={cn(
+          ' hover:text-blue-primary focus:outline-none focus:ring-4 focus:ring-blue-primary/20',
+          className,
+        )}
+        href={href}
+        prefetch={prefetch}
+        ref={ref}
+        {...rest}
+      >
+        {children}
+      </Link>
+    );
+  },
+);
```

Comment:
yes, I’ve seen it earlier  but believe that we are still better to use Link from the library since it has extended [functionality](https://next-intl-docs.vercel.app/docs/routing/navigation#link) for localisation. 

For now we’ve made a configuration `Navigation` in the same named file with [shared pathnames](https://next-intl-docs.vercel.app/docs/routing/navigation#shared-pathnames) and library provides utils for this purpose. Later, if we or catalyst’s users would like to switch to [localised pathnames](https://next-intl-docs.vercel.app/docs/routing/navigation#localized-pathnames) it’s still covered with library but in both cases they provide us their version of wrapper for `next/link` component. 

I’m thinking that maybe we can bring here( to nav-link) and kind of merge what’ve been done in `components/link` to sync them. What do you think about this?

Reply:
I think we just need to keep `Link` and instead of using `NextLink` use `Link` from library, but maybe @deini has a better approach.